Upstream: Wants it as it is.
Source: berniyh@exherbo.org
Summary: update-modules is obsolete in Exherbo and potentially harmful so we
don't use it anymore. 

diff -Naur baselayout-1.12.12.old/init.d/modules baselayout-1.12.12/init.d/modules
--- baselayout-1.12.12.old/init.d/modules	2008-08-21 20:24:03.000000000 +0200
+++ baselayout-1.12.12/init.d/modules	2008-08-21 20:28:47.000000000 +0200
@@ -76,37 +76,9 @@
 	# support compiled in ...
 	[[ -f /proc/modules ]] || return 0
 
-	# Make sure depmod from modutils do not whine, but do not bother if
-	# we are on a 2.6 kernel without modprobe.old
-	if [[ -z "${CDBOOT}" ]] && [[ ! -e /etc/modules.conf ]] && \
-	   [[ $(get_KV) -lt $(KV_to_int '2.5.48') || -x /sbin/modprobe.old ]]
-	then
-		echo '### This file is automatically generated by update-modules' \
-			> /etc/modules.conf 2>/dev/null
-		[[ ! -f /etc/modules.conf ]] && \
-			ewarn "Cannot update /etc/modules.conf!"
+	if [[ -f /etc/conf.d/modules.autoload ]]; then
+		load_modules /etc/conf.d/modules.autoload
 	fi
-	
-	# Only do this if we have modules.conf or a 2.6 kernel
-	if [[ -z "${CDBOOT}" ]] && \
-	   [[ -f /etc/modules.conf || $(get_KV) -ge $(KV_to_int '2.5.48') ]]
-	then
-		/sbin/update-modules 
-	fi
-
-	local autoload=""
-	if [[ -f /etc/modules.autoload && ! -L /etc/modules.autoload ]]; then
-		autoload=/etc/modules.autoload
-	else
-		local x
-		for x in "${KV}" ${KV_MAJOR}.${KV_MINOR}.${KV_MICRO} ${KV_MAJOR}.${KV_MINOR} ; do
-			if [[ -f /etc/modules.autoload.d/kernel-"${x}" ]] ; then
-				autoload="/etc/modules.autoload.d/kernel-${x}"
-				break
-			fi
-		done
-	fi
-	[[ -n ${autoload} ]] && load_modules "${autoload}"
 
 	#
 	# Just in case a sysadmin prefers generic symbolic links in
